pkg/cluster/fixssh.go

```diff
@@ -124,52 +153,122 @@ func (m *manager) removeBackendPoolsFromNIC(ctx context.Context, resourceGroup,
 		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
-			ID: &sshID,
+			ID: &sshBackendPoolID,
 		})
 	}
 
 	if updateILBPool {
-		m.log.Printf("Adding NIC %s to Internal Load Balancer API Address Pool %s", nicName, ilbID)
+		m.log.Printf("Adding NIC %s to Internal Load Balancer API Address Pool %s", nicName, ilbBackendPoolID)
 		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
-			ID: &ilbID,
+			ID: &ilbBackendPoolID,
 		})
 	}
 
 	return updateSSHPool || updateILBPool
 }
 
+func (m *manager) updateV1ELBAddressPool(ctx context.Context, nic *mgmtnetwork.Interface, nicName string, resourceGroup string, infraID string) error {
+	if nic.InterfacePropertiesFormat.IPConfigurations == nil || len(*nic.InterfacePropertiesFormat.IPConfigurations) == 0 {
+		return fmt.Errorf("unable to update NIC as there are no IP configurations for %s", nicName)
+	}
+
+	var updateV1ELBPool bool
+	lb, err := m.loadBalancers.Get(ctx, resourceGroup, infraID, "")
+	elbBackendPoolID := fmt.Sprintf("%s/backendAddressPools/%s", *lb.ID, infraID)
+	lbpool := *(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools
+	var index int
+	for i, p := range lbpool {
+		if strings.EqualFold(*p.ID, elbBackendPoolID) {
+			updateV1ELBPool = true
+			index = i
+		}
+	}
+
+	if updateV1ELBPool {
+		m.log.Printf("Removing NIC %s from Public Load Balancer API Address Pool %s", nicName, elbBackendPoolID)
+		lbpool = append(lbpool[:index], lbpool[index+1:]...)
+		(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = &lbpool
+		m.log.Printf("Updating Network Interface %s", nicName)
+		err = m.interfaces.CreateOrUpdateAndWait(ctx, resourceGroup, nicName, *nic)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func (m *manager) updateELBAddressPool(ctx context.Context, nic *mgmtnetwork.Interface, nicName string, lb *mgmtnetwork.LoadBalancer, i int, infraID string) bool {
+	if nic.InterfacePropertiesFormat.IPConfigurations == nil || len(*nic.InterfacePropertiesFormat.IPConfigurations) == 0 {
+		m.log.Warnf("unable to update NIC as there are no IP configurations for %s", nicName)
+		return false
+	}
+
+	var elbBackendPool string
+	switch m.doc.OpenShiftCluster.Properties.ArchitectureVersion {
+	case api.ArchitectureVersionV1:
+		elbBackendPool = infraID + "-public-lb-control-plane-v4"
+	case api.ArchitectureVersionV2:
+		elbBackendPool = infraID
+	}
+
+	elbBackendPoolID := fmt.Sprintf("%s/backendAddressPools/%s", *lb.ID, elbBackendPool)
+
+	updateELBPool := true
+	for _, p := range *(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools {
+		if strings.EqualFold(*p.ID, elbBackendPoolID) {
+			updateELBPool = false
+		}
+	}
+
+	if updateELBPool {
```

Comment:
can we make this similar to the v1 version where we do the work once we determine we need to instead of using 'tracking booleans' and pushing the logic to multiple places?

Reply:
Change the code similar to V1 version